stl/inc/span

```diff
@@ -0,0 +1,758 @@
+
+#ifdef __cpp_lib_concepts
+// clang-format off
+template<class _It, class _Ty>
+concept _Is_span_compatible_iterator = contiguous_iterator<_It>
+                                    && is_convertible_v<remove_reference_t<iter_reference_t<_It>>(*)[], _Ty(*)[]>;
+
+template<class _It, class _Sentinel>
+concept _Is_span_compatible_sentinel = sized_sentinel_for<_Sentinel, _It>
+                                    && !is_convertible_v<_Sentinel, size_t>;
+
+template <class _Range, class _Ty>
+concept _Is_span_compatible_range = !is_array_v<remove_cvref_t<_Range>>
+                                 && !_Is_span<remove_cvref_t<_Range>>::value
+                                 && !_Is_std_array<remove_cvref_t<_Range>>::value
+                                 && ranges::contiguous_range<_Range>
+                                 && ranges::sized_range<_Range>
+                                 && (ranges::forward_range<_Range> || is_const_v<_Ty>)
+                                 && is_convertible_v<remove_reference_t<ranges::range_reference_t<_Range>>(*)[], _Ty(*)[]>;
+// clang-format on
+#else // ^^^ __cpp_lib_concepts / !__cpp_lib_concepts vvv
+
+// STRUCT TEMPLATE _Is_span_compatible_range
+// clang-format off
+template <class _Range, class _Ty>
+struct _Is_span_compatible_range : bool_constant<conjunction_v<
+    negation<is_array<_Range>>,
+    negation<_Is_span<remove_const_t<_Range>>>,
+    negation<_Is_std_array<remove_const_t<_Range>>>,
+    _Has_range_interface<_Range>,
+    _Is_span_convertible_range<_Range, _Ty>>> {};
+// clang-format on
+
+#endif // !__cpp_lib_concepts
+
+// [views.span]
+// CLASS TEMPLATE span
+template <class _Ty, size_t _Extent = dynamic_extent>
+class span : public _Span_extent_type<_Extent> {
+public:
+    using _Mybase                = _Span_extent_type<_Extent>;
+    using element_type           = _Ty;
+    using value_type             = remove_cv_t<_Ty>;
+    using size_type              = size_t;
+    using difference_type        = ptrdiff_t;
+    using pointer                = _Ty*;
+    using const_pointer          = const _Ty*;
+    using reference              = _Ty&;
+    using const_reference        = const _Ty&;
+    using iterator               = _Span_iterator<_Ty>;
+    using const_iterator         = _Span_iterator<const _Ty>;
+    using reverse_iterator       = _STD reverse_iterator<iterator>;
+    using const_reverse_iterator = _STD reverse_iterator<const_iterator>;
+
+    static constexpr size_type extent = _Extent;
+
+    // [span.cons] Constructors, copy, and assignment
+    template <size_t _Ext = _Extent, enable_if_t<_Ext == 0 || _Ext == dynamic_extent, int> = 0>
+    constexpr span() noexcept {}
+
+#ifdef __cpp_lib_concepts
+
+    // clang-format off
+    template <class _It> requires _Is_span_compatible_iterator<_It, element_type>
+    constexpr span(_It _First, size_type _Count) noexcept // strengthened
+        : _Mybase(_Count), _Mydata(to_address(_First)) {

```

Comment:
Talking about `to_address`, should that be qualified via `_STD`

Reply:
Yes, you need `_STD to_address`. It's a non-`_Ugly` function, so it has to be defended against ADL hijacking.

Reply:
Done